src/anomaly/mod.rs

```diff
@@ -1 +1,11 @@
+use num::{Float, FromPrimitive};
+use std::ops::{AddAssign, DivAssign, MulAssign, SubAssign};
+
+use crate::common::{ClassifierOutput, Observation};
+
 pub mod half_space_tree;
+
+trait AnomalyDetector<F: Float + FromPrimitive + AddAssign + SubAssign + MulAssign + DivAssign> {
+    fn learn_one(&mut self, observation: &Observation<F>);
+    fn score_one(&mut self, observation: &Observation<F>) -> Option<ClassifierOutput<F>>;
```

Comment:
We continue to discuss the output type of score one, I merge your pr when we decide that is the good format. Thank you for your contribution 😄

Reply:
Aha no problem. Let me know if it's a public discussion 😃